RobotArmAPP/Classes/WiFiAPConnection.cs

```diff
@@ -0,0 +1,141 @@
+            string statusCode = response.StatusCode.ToString(); // Retorna "OK" quando da certo
+            return statusCode;
+        } //Checks if ESP32 and computer are exchanging data
+
+        public async Task<bool> GetNetworkProfiles(bool verifySSID, bool needDialog)
+        {
+            try
+            {
+                var currentNetwork = await wifiAdapter.NetworkAdapter.GetConnectedProfileAsync();
+                //await WifiStatus(false, false);
+                if (verifySSID == true)
+                {
+                    return (currentNetwork != null && currentNetwork.ProfileName == SSID);
+                }
+                else
+                {
+                    return currentNetwork != null;
+                }
+            }
+            catch
+            {
+                if (needDialog == true)
+                {
+                    var dialog = new MessageDialog("Check your WiFi network adapter!", "Error");
+                    await dialog.ShowAsync();
+                    await WifiStatus(true, false);
+                }
+                return false;
+            }
+        } //Search the networks and call the wifistatus to see if it is the right network, if yes it returns true. 
+
+        public async Task<int> WifiStatus(bool isDisconnected, bool isConnecting)
+        {
+            bool rightNetwork = await GetNetworkProfiles(true, false);
+            if (isConnecting == false)
+            {
+                if (rightNetwork == true && isDisconnected == false)
+                {
+                    try
+                    {
+                        string code = await VerifyAP(200);
+                        if (code == "OK")
+                        {
+                            status = 1;
+                        }
+                    }
+                    catch
+                    {
+                        status = 0;
+                    }
+                }
+                else
+                {
+                    status = 0;
+                }
+                return status;
+            }
+            else
+            {
+                status = 2;
+                return status;
+            }
+        } //Checks if the computer is connected to the ESP32 WiFi and call VerificarAP
+
+        public async Task ConnectToWifi()
+        {
+            try
+            {
+                var result = await Windows.Devices.Enumeration.DeviceInformation.FindAllAsync(WiFiAdapter.GetDeviceSelector());
+                if (result.Count >= 1)
+                {
+                    wifiAdapter = await WiFiAdapter.FromIdAsync(result[0].Id);
+                    await wifiAdapter.ScanAsync();// escaneando redes
+
+                    var nw = wifiAdapter.NetworkReport.AvailableNetworks.Where(y => y.Ssid == SSID).FirstOrDefault(); //verificando a rede
+                    var credential = new PasswordCredential //senha
+                    {
+                        Password = PW
```

Comment:
```suggestion
                        Password = this.Password
```
Ao usar o `this`, você está acessando a propriedade da classe

Reply:
Não funcionou!

Reply:
Não pode usar o "this" nesse caso, mas pode ficar "Password = Password", acho que fica meio confuso, mas fiz as mudanças assim mesmo.